tests/unit/test_coordinate_utilities.py

```diff
@@ -543,3 +515,246 @@ def test_any_absent_dimension_variables(self):
+                    lat_arr,
+                    lon_arr,
+                    self.varinfo.get_variable(self.latitude),
+                    self.varinfo.get_variable(self.longitude),
+                )
+                self.assertEqual(
+                    context.exception.message,
+                    'No valid coordinate data',
+                )
+        with self.subTest('no valid points in coordinates dataset'):
+            lat_arr = np.array(
+                [
+                    [-9999.0, -9999.0, -9999.0, -9999.0, -9999.0],
+                    [-9999.0, -9999.0, -9999.0, -9999.0, -9999.0],
+                ]
+            )
+            lon_arr = np.array(
+                [
+                    [-9999.0, -9999.0, -9999.0, -9999.0, -9999.0],
+                    [-9999.0, -9999.0, -9999.0, -9999.0, -9999.0],
+                ]
+            )
+            with self.assertRaises(InvalidCoordinateData) as context:
+                get_valid_row_col_pairs(
+                    lat_arr,
+                    lon_arr,
+                    self.varinfo.get_variable(self.latitude),
+                    self.varinfo.get_variable(self.longitude),
+                )
+                self.assertEqual(
+                    context.exception.message,
+                    'No valid coordinate data',
+                )
+
+    def test_get_x_y_values_from_geographic_points(self):
+        """Ensure that the correct x-values are returned by the function
+        with a set of geographic coordinates as input.
+
+        """
+        crs = CRS.from_cf(
+            {
+                'false_easting': 0.0,
+                'false_northing': 0.0,
+                'longitude_of_central_meridian': 0.0,
+                'standard_parallel': 30.0,
+                'grid_mapping_name': 'lambert_cylindrical_equal_area',
+            }
+        )
+
+        with self.subTest('Get valid x-y points from coordinates in a row'):
+            two_col_points_in_a_row = [(-179.3, 89.3), (178.4, 89.3)]
+
+            expected_x_y_points = [
+                (-17299990.048985746, 7341677.255608977),
+                (17213152.396759935, 7341677.255608977),
+            ]
+            actual_x_y_points = get_x_y_values_from_geographic_points(
+                two_col_points_in_a_row, crs
+            )
+            self.assertListEqual(actual_x_y_points, expected_x_y_points)
+
+        with self.subTest('Get valid x-y points from coordinates in a col'):
+            two_row_points_in_a_col = (-179.3, 89.3), (-179.3, -88.1)
+            expected_x_y_points = [
+                (-17299990.048985746, 7341677.255608977),
+                (-17299990.048985746, -7338157.219843731),
+            ]
+            actual_x_y_points = get_x_y_values_from_geographic_points(
+                two_row_points_in_a_col, crs
+            )
+            self.assertListEqual(actual_x_y_points, expected_x_y_points)
+
+    def test_get_dimension_array_from_geo_points(self):
```

Comment:
First up - thanks for adding unit tests for this new function!

I think the coverage of the two branches of conditional logic (x versus y) is good. I wonder, though, is it possible for the latitude array, longitude array and/or row indices have either only 1 value or 0 values? If so we probably need tests for those cases.

Reply:
Those are caught in a lower level function. But I guess I could add those here as well